config_converter/config_mapper/config_mapper.py

```diff
@@ -30,15 +30,51 @@
 _SUPPORTED_PLUGINS = ['in_tail']
 
 
+def initialize_stats(d: config_pb2.Directive) -> dict:
+    """Initializes the stats dict to print out."""
+    stats = {
+        'attributes_num': get_num_attrs(d),
+        'attributes_recognized': 0,
+        'attributes_unrecognized': 0,
+        'attributes_skipped': 0,
+        'entities_num': get_num_ents(d),
+        'entities_skipped': 0,
+        'entities_unrecognized': 0,
+        'entities_recognized_success': 0,
+        'entities_recognized_partial': 0,
+        'entities_recognized_failure': 0
+    }
+    return stats
+
+
+def get_num_attrs(d: config_pb2.Directive) -> int:
+    """Takes a directive, returns total no of attributes."""
+    num_attrs = len(list(d.params))
+    for nd in d.directives:
+        num_attrs += get_num_attrs(nd)
+    return num_attrs
+
+
+def get_num_ents(d: config_pb2.Directive) -> int:
+    """Takes a directive, returns total no of directives."""
+    num_ents = len(list(d.directives))
+    for nd in d.directives:
+        num_ents += get_num_ents(nd)
+    return num_ents
+
+
 def extract_root_dirs(config_obj: config_pb2.Directive) -> dict:
     """Checks all dirs, maps with corresponding params if supported."""
     logs_module = dict()
     result = {'logs_module': logs_module}
-    plugin_prefix_map = {'source': 'in_', 'match': 'out_'}
-    dir_name_map = {'source': 'sources', 'match': 'output'}
+    stats = initialize_stats(config_obj)
+    plugin_prefix_map = {'source': 'in_'}
```

Comment:
I removed them since we don't currently support related plugins, so when we found match directives we could count it as a skipped entity. Otherwise, it would be counted as an unknown entity which wasn't entirely correct since we are skipping such plugins on purpose by not including the name in the prefix map. Not sure if this was the right way to do it, what do you think?